bin/assigner/include/runner.hpp

```diff
@@ -0,0 +1,246 @@
+#ifndef ZKEMV_FRAMEWORK_BIN_ASSIGNER_INCLUDE_RUNNER_HPP_
+#define ZKEMV_FRAMEWORK_BIN_ASSIGNER_INCLUDE_RUNNER_HPP_
+
+#include <evmone/evmone.h>
+
+#include <boost/log/core.hpp>
+#include <boost/log/expressions.hpp>
+#include <boost/log/trivial.hpp>
+#include <nil/blueprint/assigner.hpp>
+
+#include "output_artifacts.hpp"
+#include "utils.h"
+#include "vm_host.h"
+#include "write_assignments.hpp"
+#include "zkevm_framework/data_types/account.hpp"
+#include "zkevm_framework/data_types/block.hpp"
+#include "zkevm_framework/data_types/transaction.hpp"
+
+template<typename BlueprintFieldType>
+class single_thread_runner {
+  public:
+    /// @brief Initialize runner with empty account storage
+    single_thread_runner(const std::vector<std::array<std::size_t, 4>>& column_sizes,
+                         boost::log::trivial::severity_level log_level = boost::log::trivial::info)
+        : m_column_sizes(column_sizes), m_log_level(log_level) {}
+
+    /// @brief Initialize runner predeined account storage
+    single_thread_runner(const evmc::accounts& account_storage,
+                         const std::vector<std::array<std::size_t, 4>>& column_sizes,
+                         boost::log::trivial::severity_level log_level = boost::log::trivial::info)
+        : m_account_storage(account_storage),
+          m_column_sizes(column_sizes),
+          m_log_level(log_level) {}
+
+    /// @brief Execute one block
+    int run(const unsigned char* input_block_data, size_t input_block_size,
+            const std::string& assignment_table_file_name,
+            const std::optional<OutputArtifacts>& artifacts) {
+        data_types::bytes block_data;
+        block_data.resize(input_block_size);
+        for (size_t i = 0; i < input_block_size; i++) {
+            block_data[i] = static_cast<std::byte>(input_block_data[i]);
+        }
+
+        auto maybe_input_block = data_types::Block::deserialize(block_data);
+        if (!maybe_input_block.has_value()) {
+            std::cerr << "Could not deserilize input block";
+            return -1;
+        }
+
+        auto input_block = maybe_input_block.value();
+        return run(input_block, assignment_table_file_name, artifacts);
+    }
+
+    /// @brief Execute one block
```

Comment:
Same comment as for the previous method, probably we should specify the difference

Reply:
good catch

Reply:
good catch, thanks!